app/controllers/pokemon_controller.rb

```diff
@@ -13,7 +13,13 @@ def new
   def create
     @pokemon = Pokemon.new(pokemon_params)
     @pokemon.user_id = current_user.id
-    @pokemon.species_id = Species.find_by(national_id: params[:pokemon][:species_id]).id
+    if params[:a_to_h] != ""
+      @pokemon.species_id = Species.find_by(national_id: params[:a_to_h]).id
+    elsif params[:i_to_r] != ""
+      @pokemon.species_id = Species.find_by(national_id: params[:i_to_r]).id
+    elsif params[:s_to_z] != ""
+      @pokemon.species_id = Species.find_by(national_id: params[:s_to_z]).id
```

Comment:
Why not: `@pokemon.species = Species.find_by(national_id: params[:s_to_z])` or even `@pokemon.species_id = params[:s_to_z]`